AI/BattleAI/BattleAI.cpp

```diff
@@ -197,7 +197,7 @@ BattleAction CBattleAI::useCatapult(const BattleID & battleID, const CStack * st
 	}
 	else
 	{
-		EWallPart wallParts[] = {
+		std::array<EWallPart, 7> wallParts {
```

Comment:
```suggestion
		std::array wallParts {
```
(yes, this works)
https://en.cppreference.com/w/cpp/language/class_template_argument_deduction

`std::array<EWallPart, 7>` form is actually somewhat dangerous since it does not checks for initializer size.
If you initialize such array with less than 7 elements it will still compile but won't work as expected